backend/src/test/java/com/bakdata/conquery/integration/sql/dialect/TestSqlConnectorConfig.java

```diff
@@ -5,19 +5,22 @@
 import com.bakdata.conquery.models.config.DatabaseConfig;
 import com.bakdata.conquery.models.config.SqlConnectorConfig;
 import com.bakdata.conquery.models.datasets.Dataset;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import lombok.NoArgsConstructor;
 
+@NoArgsConstructor
+@JsonDeserialize(as = TestSqlConnectorConfig.class)
 public class TestSqlConnectorConfig extends SqlConnectorConfig {
 
-	private final DatabaseConfig databaseConfig;
+	private static final String TEST_DATASET = "test";
 
 	public TestSqlConnectorConfig(DatabaseConfig databaseConfig) {
-		super(true, true, Map.of());
-		this.databaseConfig = databaseConfig;
+		super(true, true, Map.of(TEST_DATASET, databaseConfig));
 	}
 
 	@Override
 	public DatabaseConfig getDatabaseConfig(Dataset dataset) {
-		return databaseConfig;
+		return getDatabaseConfigs().get(TEST_DATASET);
```

Comment:
ist das nicht etwas gefährlich falls du Tests mit mehreren Datasets schreiben wirst?

Reply:
Dann schon, aber aktuell wird ja automatisch für pro Test ein Dataset erstellt. Damit ich darauf keine Rücksicht nehmen muss, hab ich das so gemacht. Ansonsten müsste ich das beim Erzeugen eines `IntegrationTest` setzen und da per `overrideConfig` das Mapping von `Dataset`->`DatabaseConfig` richtig setzen.

Reply:
Würde es evtl sinn machen die DB-Connection im Dataset zu verankern?

Reply:
Das ist auch was, was Max oder ich gut übernehmen können.

Reply:
@awildturtok Könnte man drüber nachdenken, aber das kannst du ggf. noch besser einschätzen als ich. Ich bin mir nicht 100% sicher, ob es in irgendwelchen Anwendungsfällen Sinn ergeben kann, eine DB-Connection für mehrere Datasets zu benutzen.

Trotzdem den PR hier erstmal mergen und die Idee nochmal als Issue aufnehmen?

Reply:
ja klingt gut